service/controller/chartconfig/v7/resource/chartstatus/create.go

```diff
@@ -29,30 +29,41 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		return microerror.Mask(err)
 	}
 
-	if key.ReleaseStatus(customObject) != releaseContent.Status {
-		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("setting status for release '%s' status to '%s'", releaseName, releaseContent.Status))
-
-		customObjectCopy := customObject.DeepCopy()
-		customObjectCopy.Status.ReleaseStatus = releaseContent.Status
+	var status, reason string
+	{
+		if key.IsCordoned(customObject) {
+			status = releaseStatusCordoned
+			reason = key.CordonReason(customObject)
+		} else {
+			status = releaseContent.Status
+			if releaseContent.Status != releaseStatusDeployed {
+				releaseHistory, err := r.helmClient.GetReleaseHistory(ctx, releaseName)
+				if err != nil {
+					return microerror.Mask(err)
+				}
 
-		if releaseContent.Status != releaseStatusDeployed {
-			releaseHistory, err := r.helmClient.GetReleaseHistory(ctx, releaseName)
-			if err != nil {
-				return microerror.Mask(err)
+				reason = releaseHistory.Description
+			} else {
+				reason = ""
 			}
-
-			customObjectCopy.Status.Reason = releaseHistory.Description
-		} else {
-			customObjectCopy.Status.Reason = ""
 		}
+	}
+
+	if key.ReleaseStatus(customObject) != status {
```

Comment:
What if the `reason` changes? Shouldn't the object be updated then?

Reply:
Good catch, Even if status is same, that would be the different reason for it.